### PR TITLE
PP-14435 add parity checker agreementPaymentType additional tests

### PR DESCRIPTION
**Pull request by hjvoid**
## WHAT YOU DID
Adding additional test for the parity checker, specifically to test AgreementPaymentType against non-matching `null` values either in connector or ledger.

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/tasks/service/ChargeParityCheckerTest.java [hunk 1]
```diff
@@ -668,6 +653,28 @@ void parityCheck_shouldReturnMatchIf3dsDataMatchesExactly(ChargeEntity chargeEnt
         ParityCheckStatus parityCheckStatus = chargeParityChecker.checkParity(chargeEntity, ledgerTransaction);
         assertThat(parityCheckStatus, is(EXISTS_IN_LEDGER));
     }
+
+    @ParameterizedTest
+    @EnumSource(AgreementPaymentType.class)
+    void parityCheck_shouldThrowForUnexpectedAgreementPaymentTypeInLedger(AgreementPaymentType agreementPaymentType) {
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/tasks/service/ChargeParityCheckerTest.java]: This name could be better: something like `shouldThrowForNonNullAgreementPaymentTypeInConnectorAndNullAgreementPaymentTypeInLedger` (aside: `Throw` kind of implies an exception so maybe `Fail` or similar would be better).

File: src/test/java/uk/gov/pay/connector/tasks/service/ChargeParityCheckerTest.java [hunk 2]
```diff
@@ -668,6 +653,28 @@ void parityCheck_shouldReturnMatchIf3dsDataMatchesExactly(ChargeEntity chargeEnt
...
+        LedgerTransaction transaction = LedgerTransactionFactory.buildTransactionWithAgreementPaymentType(chargeEntity);
+        transaction.setAgreementPaymentType(null);
+
+        ParityCheckStatus parityCheckStatus = chargeParityChecker.checkParity(chargeEntity, transaction);
+        assertThat(parityCheckStatus, is(DATA_MISMATCH));
+    }
+
+    @ParameterizedTest
+    @EnumSource(AgreementPaymentType.class)
+    void parityCheck_shouldThrowForUnexpectedAgreementPaymentTypeInConnector(AgreementPaymentType agreementPaymentType) {
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/tasks/service/ChargeParityCheckerTest.java]: Again, name could be better.

File: src/test/java/uk/gov/pay/connector/tasks/service/ChargeParityCheckerTest.java [hunk 3]
```diff
@@ -668,6 +653,28 @@ void parityCheck_shouldReturnMatchIf3dsDataMatchesExactly(ChargeEntity chargeEnt
...
+        LedgerTransaction transaction = LedgerTransactionFactory.buildTransactionWithAgreementPaymentType(chargeEntity);
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/tasks/service/ChargeParityCheckerTest.java:671]: An explicit `ChargeEntity chargeEntity = ChargeEntityFactory.createWithAgreementPaymentType(null);` might make the intent clearer here.

File: src/test/java/uk/gov/pay/connector/tasks/service/ChargeParityCheckerTest.java [hunk 4]
```diff
@@ -668,6 +653,28 @@ void parityCheck_shouldReturnMatchIf3dsDataMatchesExactly(ChargeEntity chargeEnt
...
+        transaction.setAgreementPaymentType(agreementPaymentType);
+
+        ParityCheckStatus parityCheckStatus = chargeParityChecker.checkParity(chargeEntity, transaction);
+        assertThat(parityCheckStatus, is(DATA_MISMATCH));
+    }
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/tasks/service/ChargeParityCheckerTest.java:676]: We could do with making `parityChecker_shouldThrowForNonMatchingAgreementPaymentTypeFields(…)` more exhaustive by comparing every possible connector value against every possible ledger value except the matching one. A `@MethodSource` like this would work:

```java
private static List<Arguments> parityChecker_shouldThrowForNonMatchingAgreementPaymentTypeFields() {
    List<Arguments> arguments = new ArrayList<>();
    for (var connectorType : AgreementPaymentType.values()) {
        for (var ledgerType : AgreementPaymentType.values()) {
            if (connectorType != ledgerType) {
                arguments.add(Arguments.of(connectorType, ledgerType));
            }
        }
    }
    return List.copyOf(arguments);
}
```

**Review by alexbishop1 — CHANGES_REQUESTED** [review]

**Comment by hjvoid** [comment]
I've made the adjustments as well as grouped the AgreementPaymentType tests together as they were a bit jumbled about beforehand.

**Review by alexbishop1 — APPROVED** [review]
